src/main/java/io/cryostat/recordings/RecordingTargetHelper.java

```diff
@@ -156,6 +172,115 @@ public IRecordingDescriptor startRecording(
+        try {
+            Optional<InputStream> recording =
+                    targetConnectionManager.executeConnectedTask(
+                            connectionDescriptor,
+                            conn ->
+                                    conn.getService().getAvailableRecordings().stream()
+                                            .filter(r -> Objects.equals(recordingName, r.getName()))
+                                            .map(
+                                                    desc -> {
+                                                        try {
+                                                            return conn.getService()
+                                                                    .openStream(desc, false);
+                                                        } catch (Exception e) {
+                                                            logger.error(e);
+                                                            return null;
+                                                        }
+                                                    })
+                                            .filter(Objects::nonNull)
+                                            .findFirst());
+
+            future.complete(recording);
+        } catch (Exception e) {
+            future.completeExceptionally(e);
+        }
+        return future;
+    }
+
+    public Future<Void> deleteRecording(
+            ConnectionDescriptor connectionDescriptor, String recordingName) {
+        CompletableFuture<Void> future = new CompletableFuture<>();
+        try {
+            String targetId = connectionDescriptor.getTargetId();
+            targetConnectionManager.executeConnectedTask(
+                    connectionDescriptor,
+                    connection -> {
+                        Optional<IRecordingDescriptor> descriptor =
+                                connection.getService().getAvailableRecordings().stream()
+                                        .filter(
+                                                recording ->
+                                                        recording.getName().equals(recordingName))
+                                        .findFirst();
+                        if (descriptor.isPresent()) {
+                            connection.getService().close(descriptor.get());
+                            reportService.delete(connectionDescriptor, recordingName);
+                        } else {
+                            throw new RecordingNotFoundException(targetId, recordingName);
+                        }
+                        return null;
+                    });
+            future.complete(null);
+        } catch (Exception e) {
+            future.completeExceptionally(e);
+        }
+        return future;
+    }
+
+    public Future<SnapshotMinimalDescriptor> createSnapshot(JFRConnection connection) {
```

Comment:
For consistency with the other methods like `startRecording`, `getRecording`, `deleteRecording`, should this method take a `ConnectionDescriptor` parameter instead of a `JFRConnection`? Then the current method body would basically just get wrapped in a lambda and passed to `targetConnectionManager.executeConnectedTask` like the other methods do.

Reply:
The reason I passed in a `JFRConnection` instead is because that the `v1` and `v2` `TargetSnapshotPostHandler` implementations of the `targetConnectionManager.executeConnectedTask` both return different objects (`String` vs. `SnapshotDescriptor`, respectively) with the `v2` version also making using of the `connection` lambda parameter inside the returned `SnapshotDescriptor` object. That's why I decided to just extract the maximally mutual code between the two.

It's a bit hard to explain but if you look at both versions maybe you have a better idea of how I could go about this with a `ConnectionDescriptor` parameter instead?

Reply:
I think it would be a slightly heavier refactor, but the `SnapshotMinimalDescriptor` seems like it can be removed, too. If you opened a target connection in this method, created the snapshot, renamed it, and then queried again for recordings matching the new rename and returned that `IRecordingDescriptor`, would that fit the needs of both V1 and V2 handlers? V1 would just return the name from that descriptor since that is already the rename, and V2 would use that same name plus the rest of the whole descriptor (no more need for V2's own `SnapshotDescriptor` that overrides the name field).